src/csharp/Grpc.Tools/Common.cs

```diff
@@ -87,6 +87,16 @@ static Platform()
 
             // Hope we are not building on ARM under Xamarin!
             Cpu = Environment.Is64BitOperatingSystem ? CpuKind.X64 : CpuKind.X86;
+
+            // See https://github.com/grpc/grpc/issues/18543
+            // Problem: Is64BitOperatingSystem returns "true" inside a 32-bit docker container.
```

Comment:
I am wondering what's the harm in just invoking 32 bit protoc if the current process is 32 bit, and vice-versa if 64 bit. AFAIK there shouldn't be any behavioral difference between the 32 and 64 bit versions of protoc and the plugin.

Reply:
My analysis:
- On windows that should be fine because you can run both 32bit and 64bit protoc.exe and they should work.
- On macos (where 32bit mono used to be the default and then it changed to 64bit) that should work as well because on MacOS running a 32bit binary also should work (I'll test that).
- On linux that works too because even though running a 32bit binary on a 64bit linux system generally won't work (libraries will be missing), having 32bit mono seem to be enough evidence to assume we can safely run a 32bit protoc.

Reply:
thanks, AFAICS too the concern in the last bullet is already a problem with the current fix

Reply:
I switched from OSArchitecture to ProcessArchitecture because that should work fine based on the reasoning above (and it also simplifies the logic).